File: Seccion 18 - Clases en C++/Seccion_18_Dart/01_ejercicio_01.dart

import 'dart:io';

class DiaAnio {

  int _dia, _mes;

  DiaAnio(this._dia, this._mes);

  bool igual(DiaAnio d) {
    if (_dia == d._dia && _mes == d._mes){
      return true;
    } else {
      return false;
    }
  }

  void visualizar() {
    print("\nMostrando los datos:");
    print("Mes: $_mes\nDía: $_dia");
  }
}

void main() {

  DiaAnio hoy, cumple;
  int dia, mes;

  stdout.write("Introduzca la fecha actual, día: ");
  dia = int.parse(stdin.readLineSync());
  stdout.write("Mes: ");
  mes = int.parse(stdin.readLineSync());

  hoy = new DiaAnio(dia, mes);

  stdout.write("Introduzca la fecha de su cumpleaños, día: ");
  dia = int.parse(stdin.readLineSync());
  stdout.write("Mes: ");
  mes = int.parse(stdin.readLineSync());

  cumple = new DiaAnio(dia, mes);

  // Mostrando las fechas
  hoy.visualizar();
  print("");
  cumple.visualizar();

  if (hoy.igual(cumple)) {
    print("\nFeliz cumpleaños!!");
  } else {
    print("\nQue tenga un buen día!!");
  }
}

File: Seccion 18 - Clases en C++/Seccion_18_Python/01_ejercicio_01.py

class DiaAnio:
    
    _dia: int
    _mes: int
    
    def __init__(self, dia, mes):
        self._dia = dia
        self._mes = mes
    
    def igual(self, d):
        if self._dia == d._dia and self._mes == d._mes:
            return True
        else:
            return False
    
    def visualizar(self):
        print("\nMostrando los datos:")
        print(f"Mes: {self._mes}\nDia: {self._dia}")

def main():
    
    dia = int(input("Introduzca la fecha actual, día: "))
    mes = int(input("Mes: "))
    
    hoy = DiaAnio(dia, mes)
    
    dia = int(input("Introduzca la fecha de su cumpleaños, día: "))
    mes = int(input("Mes: "))
    
    cumple = DiaAnio(dia, mes)
    
    # Mostramos la fecha actual
    hoy.visualizar()
    print()
    # Mostramos la fecha del cumpleaños
    cumple.visualizar()
    
    if hoy.igual(cumple):
        print("\nFeliz cumpleaños!!")
    else:
        print("\nQue tenga un buen día!!")

if __name__ == "__main__":
    main()